docs/userguide/quickstart.rst

```diff
@@ -442,14 +442,24 @@ distribution so others can use it. This functionality is provided by
 <PyPUG:tutorials/packaging-projects>`.
 
 
-Transitioning from ``setup.py`` to ``setup.cfg``
-------------------------------------------------
+Transitioning from ``setup.py`` to declarative config
+-----------------------------------------------------
 To avoid executing arbitrary scripts and boilerplate code, we are transitioning
-into a full-fledged ``setup.cfg`` to declare your package information instead
-of running ``setup()``. This inevitably brings challenges due to a different
-syntax. :doc:`Here </userguide/declarative_config>` we provide a quick guide to
-understanding how ``setup.cfg`` is parsed by ``setuptools`` to ease the pain of
-transition.
+from defining all your package information by running ``setup()`` to doing this
+declaratively - using ``setup.cfg`` or ``pyproject.toml``.
+
+To ease the challenges of transitioning, :doc:`here </userguide/declarative_config>`
+we provide a quick guide to understanding how ``setup.cfg`` is parsed by
+``setuptools``. Alternatively, :doc:`here </userguide/pyproject_config>` is the
+guide for ``pyproject.toml``.
+
+.. note::
+
+    The approach ``setuptools`` would like to take is to eventually use a single
+    declarative format (``pyproject.toml``) instead of maintaining 2
+    (``pyproject.toml`` / ``setup.cfg``). Chances are, ``setup.cfg`` will
+    continue to be maintained for a long time. Currently, ``pyproject.toml`` may 
+    still have some `limitations <https://github.com/pypa/setuptools/issues/3683>`_.
```

Comment:
Thank you for clarifications!

The 2 paragraphs seem to contradict one another. I think the limitations of `pyproject.toml` are NOT the same (but slightly higher) than those of `setup.cfg` for setuptools configurations - because of difficulty with `dynamic`.

(or is `dynamic` not related to setuptools configuration?)


But I am confused: I though that [here](https://github.com/pypa/setuptools/issues/3683#issuecomment-1380668530) you meant there shouldn't be a recommendation between `pyproject.toml` and `setuf.cfg` BECAUSE "`pyproject.toml` imposes constraints around dynamic".

If `setuf.cfg`

Reply:
But the point is that the extra faff around `dynamic` in `pyproject.toml` is not a limitation. It is a different way of writing the same thing.

You can express the same thing in `pyproject.toml` and `setup.cfg`. The end result is the same, what changes is the `how`.

Reply:
So what's left is to rephrase this line:

> Currently, ``pyproject.toml`` may 
    still have some `limitations <https://github.com/pypa/setuptools/issues/3683>`_.

Do you have any good suggestions in mind?

Reply:
Umm... tough one, maybe something like the following:

> ```rst
> Please note that `pyproject.toml` requires extra consideration regarding dynamic configs,
> see the :doc:`PyPUG:guides/writing-pyproject-toml` guide for more information.
> ```

Although this might be a bit too detailed for the quickstart, probably more appropriate for the dedicated page on `pyproject.toml`?

Reply:
I also thought to just delete the last sentence

Reply:
The PyPUG page doesn't describe this "extra consideration".. On the contrary, it says:
> When a field is dynamic, it is the build backend’s responsibility to fill it. Consult your build backend’s documentation to learn how it does it.

and the setuptools guide does have a larger section "Dynamic Metadata" already👍